docs/source/user-guide/vision/supported_tasks_and_formats.rst

```diff
@@ -0,0 +1,124 @@
+sample has 2 bounding boxes, should be provided as follows:
+
+.. code-block:: python
+
+    [
+        [(1, 50, 20, 9.4, 5.5)],
+        [(3, 12, 23, 3.4, 9.5), (5, 42, 63, 9.4, 11.5)]
+    ]
+
+Prediction Format
+-----------------
+Object detection prediction per sample should be an iterable of predicted bounding boxes and their corresponding
+confidence provided by the model. Each bounding box should be an iterable of 6 elements in the following order:
+``(x_min, y_min, w, h, confidence, class_id)``.
+
+    ``x_min``, ``y_min``, ``w`` and ``h`` represent the bounding box location as above,
+    ``confidence`` is the confidence score given by
+    the model to bounding box and ``class_id`` is the class id predicted by the model.
+
+For example, predictions of a batch containing 2 samples, where the first sample has two predicted bounding boxes and
+the second one has no predicted bounding boxes, should be provided as follows:
+
+.. code-block:: python
+
+    [
+        [(50, 20, 9.4, 5.5, 0.8, 3), (50, 20, 9.4, 5.5, 0.4, 1)],
+        []
+    ]
+
+.. _supported_tasks__segmentation:
+Semantic Segmentation
+=====================
+Label Format
+------------
+Semantic segmentation label per sample should be an array of shape ``[H, W]`` where ``H`` and ``W`` are the
+height and width of the corresponding image, and its values are the true class_ids of
+the corresponding pixels in that image.
+Note that the array should be 2D, as the number of channels on the original image are irrelevant to the label.
+
+Prediction Format
+-----------------
+Semantic segmentation prediction per sample should be **predicted probabilities** per class per pixel. Specifically,
+the prediction per sample format is an array of shape ``[C, H, W]`` where ``H`` and ``W`` are the height
+and width of the corresponding image, and ``C`` is the number of possible classes
+such that each channel corresponds to a class_id.
+
+Note that the values of dimension C are the probabilities for each class and as such for each pixel the sum of the
+probabilities should be 1.
+
+Other Tasks
+===========
+For other tasks, the label and prediction formats are not validated nor used by the different checks.
+However, there are many checks that require only the images themself. Few examples for such checks include:
+:doc:`Image Property Outliers </checks_gallery/vision/data_integrity/plot_image_property_outliers.html>`,
+:doc:`Image Dataset Drift </checks_gallery/vision/train_test_validation/plot_image_dataset_drift.html>` and
+:doc:`Image Property Drift </checks_gallery/vision/train_test_validation/plot_image_property_drift.html>`.
+
+In addition it is possible to generate custom metrics and properties for the predictions and labels
+provided and run additional checks. For more information on how to do so, see the
+:doc:`custom task tutorial </user-guide/vision/tutorials/custom_task_tutorial>`.
```

Comment:
Would maybe rephrase to something that encourages action, such as:

For "other" tasks, you can still use all of the deepchecks checks and suites! However, for some checks you will need to define a specific performance metric and label properties, as detailed in the ..."

And maybe put this before saying which checks do work?